### PR TITLE
BioSpheres by Kadax, TheCello, TheViola, and JackofLava

### PR DESCRIPTION
**Pull request by JackofLava**

### CODE REVIEW

**Comment by Jo5629** [comment]
I think am not the person that should be saying it but:
1. The title for the PR is wrong.
2. You need to rename the screenshots and make sure it is in 3:2 form.

**Comment by JackofLava** [comment]
Thank you for pointing that out, though I don't know how to do a 3:2 form
screenshot

**Comment by LuismiInTheZone** [comment]
> I think am not the person that should be saying it but:
> 
> 1. The title for the PR is wrong.
> 2. You need to rename ONE screenshot to screenshot.png and make sure it is in 3:2 form.
>    A link to help you: https://github.com/MT-CTF/capturetheflag/tree/master/mods/ctf/ctf_map#ctf-map---map-maker

The name is ok, I don't see any problem there, also what you said, you need to rename the screenshot screenshot.png, also I think that with one screenshot is enough

**Comment by LuismiInTheZone** [comment]
> > I think am not the person that should be saying it but:
> > 
> > 1. The title for the PR is wrong.
> > 2. You need to rename ONE screenshot to screenshot.png and make sure it is in 3:2 form.
> >    A link to help you: https://github.com/MT-CTF/capturetheflag/tree/master/mods/ctf/ctf_map#ctf-map---map-maker
> 
> The name is ok, I don't see any problem there, also what you said, you need to rename the screenshot screenshot.png, also I think that with one screenshot is enough

Also would be much better if you put the map in singlenode, do /ctf_start and take a screenshot without barrier

**Comment by JackofLava** [comment]
Then how do I change a file name that has been uploaded?

**Comment by LuismiInTheZone** [comment]
> Then how do I change a file name that has been uploaded? Also I trying to delete the bottom screenshot

You can or: 1. Open the .conf file in your computer and change it, or 2. Edit it from github, there's an option to do that when you go to the file

**Comment by src4026** [comment]
IIRC it isn't possible to rename image files from the GitHub website. You'll need to delete that file and add it back but making sure that the file you are uploading is the right name. Also, you seem to want to delete the screenshot but can't find the option, so here it is: (that's what you meant right?)

**Comment by src4026** [comment]
Wait, when did Luis comment? Network is behaving weird

**Comment by JackofLava** [comment]
I can't seem to edit or delete the existing screenshots that have been uploaded for some reason. Would someone, like HobbitPower, in the management/administration of the maps be able to delete or edit the existing screenshots on their end?

**Comment by JackofLava** [comment]
Let me try again

**Comment by JackofLava** [comment]
ok this is my problem, the edit button on for both screenshots is greyed out and I can't use then at all. This is under "File Changes."

**Comment by src4026** [comment]
Try going to your fork of the repository and deleting the files from there.

**Comment by MoreKat** [comment]
> Try going to your fork of the repository and deleting the files from there.

Yes this is what you need to do. Any changes you make to your master branch will appear in this PR. This is also why it's best to open a separate branch for each PR you make. Like for this one you could've created a new branch called 'biospheres', if you then decide to make another PR for another map you can create a new branch off of your master branch (which you cannot do now because it will add onto this PR).

**Comment by src4026** [comment]
Hmm. I don't remember if I've added that to the README or not.

**Comment by HobbitPower** [comment]
> I can't seem to edit or delete the existing screenshots that have been uploaded for some reason. Would someone, like HobbitPower, in the management/administration of the maps be able to delete or edit the existing screenshots on their end?

@JackofLava I can delete screenshots there so can you ;) I already deleted one of the screenshots, so you can try to delete the other one:
Go to https://github.com/JackofLava/maps/tree/master/BioSperes click your file click the three dots and "delete" file like in the picture
image
Then click "commit changes" and confirm it with clicking "commit changes" again.
image

Now you can add a new file. Take a new screenshot or take this for example. It must be named screenshot.png and should show no red barriers, no HUD, chat or coordinates. Also it must have a 3:2 ratio.
screenshot


Also I saw that the name of your folder is "BioSperes" instead of "bio_spheres". I don't know if that's a problem, usually the folder name of the maps is written in lower case. (@LoneWolfHT)

**Comment by LoneWolfHT** [comment]
`bio_spheres` would be preferable

**Comment by HobbitPower** [comment]
> `bio_spheres` would be preferable

@JackofLava when you are using the Website version of Github: go to your [fork](https://github.com/JackofLava/maps/tree/master/BioSperes), press `.` on your keyboard. That way the web editor will open. Search for `BioSpheres` in the list on the left. Select this folder and right click on it. Select rename and then rename the folder name to `bio_spheres`. Then switch to Source control (this icon). 
image

In the field you have to type a short message (Name change for example) and then you can click Commit & Push (the green field). 

In my test branch this worked at least.

And again please change your screenshot. Delete the current one and upload a new one without the red barriers. I've also provided two for you to use if you'd like.
Remember that the name of the file must be screenshot.png before you upload it.

**Comment by farooqkz** [comment]
This map indeed seems interesting :)

**Comment by HobbitPower** [comment]
@JackofLava I would just like to remind you that there are requests for changes.
So if you are still interested in the map, please let us know.
There is also a new CTF Map Editor version, so the map should be updated anyway. You will need the latest version of the CTF subgame, which can be found in the Minetest Content Database.

**Comment by JackofLava** [comment]
Ok, thank you for letting me know, I don't know when I will be able to get
to it though.

**Comment by olliy1or** [comment]
@JackofLava when the requested changes are applied, request for review

**Comment by ClothierEdward** [comment]
I really like the idea for this map. The idea of having multiple biomes in the sky is quite the unique and cool idea for a map. It's also a four-team map which I love to see. I also like that there's five biospheres for five different biomes: forest, savanna, coniferous forest, jungle, and ocean. 
I only have three criticisms for this map.
1. The name has a typo when you search for the map using /maps. It's called "BioSperes" instead of "BioSpheres".
2. You can't reach the biospheres with all the chests without bridging. I find it weird that there's already bridges to get to each team base but no bridges to get to the chests.
3. The green base (the one with the taiga biosphere) needs some additional defenses It's strange that all the other team bases have some defenses (red has a castle, the purple flag is hanging out in the trees, and blue has a wooden house) but yet green team gets nothing? The green base should have something like some ruins or a wooden house similar to blue team.
Fix those issues and you have a good map on your hands.

**Comment by Themostrandom** [comment]
image
image
image

Hello, on the whole, I like your map, but the flag platform on the jungle sphere should be enlarged, and the holes in the flag sphere (all of them) should be plugged. The ores are badly repaired 16-17-24. The base in the fir biome is less protected than the others (no houses or anything).

**Comment by a-blob** [comment]
Continued by #380
